### ISSUE TITLE
Assigning an incompatible type to a self attribute does not report an error

### ISSUE THREAD

**Issue by dsuch** [python/mypy#9178]
Hello,

* Are you reporting a bug, or opening a feature request?

I am not sure if the behaviour below is to be expected.

* Please insert below the code you are checking with mypy.

```python
class MyClass:
    abc = 123 # type: list
    def __init__(self):
        self.zxc = 456 # type: list
```

Or:

```python
class MyClass:
    abc:list = 123
    def __init__(self):
        self.zxc:list = 456
```

* What is the actual behavior/output?

```bash
$ mypy mypy2.py 
mypy2.py:2: error: Incompatible types in assignment (expression has type "int", variable has type "List[Any]")
Found 1 error in 1 file (checked 1 source file)
$
```

* What is the behavior/output you expect?

I thought a type error would be reported for an assignment to self.zxc as well.

* What are the versions of mypy and Python you are using?

Python 3.6.9 and mypy 0.790+dev.28829fbb684d7535680934b401c05698db5b4d85.

Thanks.

**Comment by dsuch** [comment]
I have found the answer.

No signatures with type declarations means no type checks, no matter if type hints exist anywhere in a function or class, unless --strict or --disallow-untyped-defs flags are used.

In particular, [this section](https://mypy.readthedocs.io/en/latest/class_basics.html#annotating-init-methods) explains it for ```__init__``` methods.
